db/index.go

```diff
@@ -381,6 +381,11 @@ func (index *collectionUniqueIndex) Update(
 	oldDoc *client.Document,
 	newDoc *client.Document,
 ) error {
+	// We only need to update the index if one of the indexed fields
+	// on the document has been changed.
+	if !isUpdatingIndexedFields(index, oldDoc, newDoc) {
```

Comment:
suggestion: I think it also makes sense to do this check for regular indexes otherwise they will do unnecessary work with storage.

Reply:
I'll open a new PR to cover that case 👍